parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -133,9 +132,13 @@ def __init__(
         self.json_ytree = json_ytree
         self._trees: dict[str, dict[str, str]] = defaultdict(dict)
         self._unavailable_modules = []
-        LOGGER.info('get all existing modules')
-        response = requests.get('{}/search/modules'.format(self._yangcatalog_api_prefix), headers=json_headers)
-        existing_modules = response.json().get('module', [])
+
+        LOGGER.info('Fetching all existing modules.')
+        existing_modules = fetch_modules(LOGGER)
+        if existing_modules is None:
+            LOGGER.error('Failed to fetch modules from API.')
```

Comment:
This message seems to be duplicated everywhere where `fetch_modules()` is called. Instead of returning an optional, it would probably be better if `fetch_modules()` raised a `RuntimeError` with this message if it fails.